_posts/2015-04-18-fsharp-core-notes.md

```diff
@@ -75,7 +75,23 @@ If you are using an old-style project file, see the information further below.
 
 If using new-style .NET SDK project files, use:
 
-    <PackageReference Update="FSharp.Core" Version="4.2.3" />
+    <PackageReference Update="FSharp.Core" Version="4.5.0" />
+
+### Make your FSharp.Core references explicit
+
+Modern .NET SDK-style F# projects allow for an _implicit_ FSharp.Core reference, where the `FSharp.Core` package referenced is determined by the .NET SDK being used to compile the project.  The exact version referenced will depend on your .NET SDK, e.g. see [this table](https://github.com/fsharp/fsharp.github.io/pull/89#issuecomment-426729015).
+
+Implicit FSharp.Core references only apply to F# projects and not C# projects. 
+
+In most software engineering it is better to pin down the exact version of FSharp.Core being used. This means your referenced libraries don't depend on tooling version. This is particularly important when making nuget packages where you don't want to unexpectedly force a package upgrade on downstream consumers.  It is also important when working with mixed F#/C# solutions.
```

Comment:
I don't see the F#/C# connection here. C# will respect whatever is specified in an F# project. If I specify nothing explicitly, the SDK version will be referenced. If I specify my own version, C# will respect that. As mentioned in another comment, this may _not_ have been the case prior to fixing a bug in our targets, where the bug fix shipped in SDK 2.1.400.

Reply:
There is also a downside to mention here. If made explicit, this will potentially result in confusing compile errors in the future, should the underlying compiler by updated and a feature be used that depends on a newer FSharp.Core version.

A concrete example of this in another language was C# 7.0 tuple support, where the majority of users who tried the feature were given an error saying that they didn't have System.ValueTuple. This caused massive confusion, since people assumed they could just upgrade their tools (which came with a new compiler) and start using the new feature. We communicate things the same way for F# (use F# 4.5 and get new byref stuff!), so it stands to reason that people will be confused by this if they stick with an explicit reference and aren't as savvy about how language features surface as we are.

This is at the root of our disagreement on the VF# repo, and I think it's important to mention as the tradeoff being made here.

Reply:
> There is also a downside to mention here. If made explicit, this will potentially result in confusing compile errors in the future, should the underlying compiler be updated and a feature be used that depends on a newer FSharp.Core version.

That should be fixed in the compiler, not via implcit references